airflow/dag_processing/manager.py

```diff
@@ -1029,6 +1051,25 @@ def _collect_results_from_processor(self, processor) -> None:
         )
         self._file_stats[processor.file_path] = stat
         file_name = Path(processor.file_path).stem
+
+        """crude exposure of instrumentation code which may need to be furnished"""
+        span = Trace.get_tracer("DagFileProcessorManager").start_span(
+            "dag_processing", start_time=datetime_to_nano(processor.start_time)
```

Comment:
Is it possible for tracers to just accept datetime objects instead? From past experience with StatsD, using a raw number as value is prone to user errors, especially since Airflow does not otherwise use nanoseconds, but either milliseconds or seconds instead. Having two time units is already one too many, adding a third one isn’t a good idea.

Reply:
Unfortunately, it is not possible, as OTEL's time unit is not compatible with datetime type.